### PR TITLE
gnrc: ipv6: netif: documentation update

### PR DESCRIPTION
**Pull request by OlegHahm**
Added a hint to router advertisements to the IPv6 netif flag documentation.

### CODE REVIEW

**Comment by OlegHahm** [comment]
Did I get this right?

File: sys/include/net/gnrc/ipv6/netif.h [hunk 1]
```diff
@@ -114,6 +114,13 @@ extern "C" {
 /**
  * @{
  * @name Flags for the interfaces
+ *
+ * @note The most-significant byte of the these flags is identical to the flags
+ *       in IPv6 router advertisements. See <a
+ *       href="https://tools.ietf.org/html/rfc4861#section-4.2">RFC4861,
+ *        section 4.2</a> and <a
+ *       href="https://tools.ietf.org/html/rfc6275#section-7.1">RFC6275,
```
Comment (authmillenon) [inline sys/include/net/gnrc/ipv6/netif.h]: If you are so diligent you can also add https://tools.ietf.org/html/rfc4191#section-2.2 and https://tools.ietf.org/html/rfc4389#section-4.1.3.3 :-)

Comment (authmillenon) [inline sys/include/net/gnrc/ipv6/netif.h]: How? In RFC4861 only M and O are defined and they are at the same position.

Comment (OlegHahm) [inline sys/include/net/gnrc/ipv6/netif.h]: But the rest is defined as "reserved". I always thought that this keyword is equivalent to "not defined yet".

Comment (authmillenon) [inline sys/include/net/gnrc/ipv6/netif.h]: Yes, and the other RFCs define them, so were do you see the contradiction?

Comment (authmillenon) [inline sys/include/net/gnrc/ipv6/netif.h]: (The other PRs are older, because RFC 4861 obsoletes RFC 2461)

Comment (OlegHahm) [inline sys/include/net/gnrc/ipv6/netif.h]: Hence, "reserved" is not equivalent to "not defined yet"?

Comment (authmillenon) [inline sys/include/net/gnrc/ipv6/netif.h]: Now I'm confused… what do you mean? Those flags were not defined when RFC 2461 was published.

Comment (authmillenon) [inline sys/include/net/gnrc/ipv6/netif.h]: Just to give further credence to the flags: RFC 4191 is even [referenced in RFC 6775](https://tools.ietf.org/html/rfc6775#section-6)

Comment (OlegHahm) [inline sys/include/net/gnrc/ipv6/netif.h]: Yes, but in RFC 4861 only the `M` and `O` bit are mentioned, while the older RFCs additionally define `Prf` and `P`.

File: sys/include/net/gnrc/ipv6/netif.h [hunk 2]
```diff
@@ -114,6 +114,13 @@ extern "C" {
 /**
  * @{
  * @name Flags for the interfaces
+ *
+ * @note The most-significant byte of the these flags is identical to the flags
```
Comment (cgundogan) [inline sys/include/net/gnrc/ipv6/netif.h]: irrespective of the below discussion: strikethrough the needless ~~the~~

File: sys/include/net/gnrc/ipv6/netif.h (hunk 1 above)
Comment (authmillenon) [inline sys/include/net/gnrc/ipv6/netif.h]: I think that was done since they don't are of relevance to the document.

Comment (OlegHahm) [inline sys/include/net/gnrc/ipv6/netif.h]: I wonder if at the end a link to just https://tools.ietf.org/html/rfc4389#section-4.1.3.3 would have been enough. But anyway, better than nothing.

**Comment by OlegHahm** [comment]
Addressed comments.

File: sys/include/net/gnrc/ipv6/netif.h [hunk 3]
```diff
@@ -114,6 +114,17 @@ extern "C" {
...
+ * @note The most-significant byte of these flags is identical to the flags in
+ *       IPv6 router advertisements. See <a
+ *       href="https://tools.ietf.org/html/rfc4861#section-4.2">RFC4861,
+ *       section 4.2</a>, <a
+ *       href="https://tools.ietf.org/html/rfc6275#section-7.1">RFC6275,
+ *       section 7.1</a>, <a
+ *       href="https://tools.ietf.org/html/rfc4191#section-2.2">RFC4191,
+ *       section 2.2</a>, and <a
+ *       href="https://tools.ietf.org/html/rfc4389#section-4.1.3.3">RFC4389,
+ *       section 2.2</a>.
```
Comment (authmillenon) [inline sys/include/net/gnrc/ipv6/netif.h]: s/2.2/4.1.3.3/g

**Comment by authmillenon** [comment]
ACK, please squash

**Comment by OlegHahm** [comment]
squashed

**Pull request merged**